addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1107 +1,1031 @@
-            tooltipItems: tooltipItems,
-            maxWidth: maxTooltipLabelWidth,
-        })).css({top: '2px', left: '2px'});
-        const $container = this.$el.find('.o_graph_canvas_container');
-        $container.append(this.$tooltip);
-
-        let top;
-        const tooltipHeight = this.$tooltip[0].clientHeight;
-        const minTopAllowed = Math.floor(chartAreaTop);
-        const maxTopAllowed = Math.floor(window.innerHeight - rendererTop - tooltipHeight) - 2;
-        const y = Math.floor(tooltipModel.y);
-        if (minTopAllowed <= maxTopAllowed) {
-            // Here we know that the full tooltip can fit in the screen.
-            // We put it in the position where Chart.js would put it
-            // if two conditions are respected:
-            //  1: the tooltip is not cut (because we know it is possible to not cut it)
-            //  2: the tooltip does not hide the legend.
-            // If it is not possible to use the Chart.js proposition (y)
-            // we use the best approximated value.
-            if (y <= maxTopAllowed) {
-                if (y >= minTopAllowed) {
-                    top = y;
+        /**
+         * Compute various objects based on the param props.
+         * @param {Object} props
+         */
+        _computeDerivedProps(props = this.props) {
+            const filteredDataPoints = this._filterDataPoints(props);
+            this.processedDataPoints = this._sortDataPoints(filteredDataPoints, props);
+
+            this._setNoContentHelper(props);
+
+            if (!this.noContentHelperData && props.comparisonFieldIndex === 0) {
+                this.dateClasses = this._getDateClasses(this.processedDataPoints, props);
+            }
+        }
+
+        /**
+         * This function creates a custom HTML tooltip.
+         * @private
+         * @param {Object} tooltipModel see chartjs documentation
+         */
+        _customTooltip(tooltipModel) {
+            this.el.style.cursor = 'default';
+            if (this.tooltip) {
+                this.tooltip.remove();
+            }
+            if (tooltipModel.opacity === 0) {
+                return;
+            }
+            if (tooltipModel.dataPoints.length === 0) {
+                return;
+            }
+            if (this._isRedirectionEnabled()) {
+                this.el.style.cursor = 'pointer';
+            }
+
+            const chartArea = this.chart.chartArea;
+            const chartAreaLeft = chartArea.left;
+            const chartAreaRight = chartArea.right;
+            const chartAreaTop = chartArea.top;
+            const rendererTop = this.el.getBoundingClientRect().top;
+
+            const maxTooltipLabelWidth = Math.floor((chartAreaRight - chartAreaLeft) / 1.618) + 'px';
+            const tooltipItems = this._getTooltipItems(tooltipModel);
+
+            const template = document.createElement('template');
+            const tooltipHtml = qweb.render('GraphView.CustomTooltip', {
```

Comment:
should use this.env.qweb (and remove require)

Reply:
this is not the owl qweb

Reply:
Why? core.qweb is meant to disappear at some point

Reply:
I now use the owl qweb and an owl template